src/ir/possible-contents.cpp

```diff
@@ -1384,6 +1389,34 @@ struct Flower {
   // We will need subtypes during the flow, so compute them once ahead of time.
   std::unique_ptr<SubTypes> subTypes;
 
+  // The depth of children for each type. This is 0 if the type has no
+  // subtypes, 1 if it has subtypes but none of those have subtypes themselves,
+  // and so forth.
+  std::unordered_map<HeapType, Index> maxDepths;
+
+  // Given a ConeType, normalize it, that is, make its depth the canonical
+  // depth given the actual children it has. If this is a full cone, then we can
+  // always pick the actual maximal depth and use that instead of FullDepth==-1.
+  // For a non-full cone, we also reduce the depth as much as possible, so it is
+  // equal to the maximum depth of an existing subtype.
+  void normalizeConeType(Type type, Index& depth) {
```

Comment:
It seems simpler to return the normalized depth.

Reply:
Good idea, done. Renamed to `getNormalizedConeDepth`.